client/dom/svg.legend.js

```diff
@@ -189,21 +189,30 @@ export default function svgLegend(opts) {
 		let colorGradientId
 		if (d.domain) {
 			colorGradientId = `sjpp-linear-gradient-${getId()}`
-			new ColorScale({
+			const data = d.domain || [d.minLabel, d.maxLabel]
```

Comment:
If `data` is used to compute scales or other values, then the defaults of using `minLabel` and `maxLabel` for `data` is risky, since a label can contain string like `0.1 mg`. Should use a data property that is guaranteed to be raw numeric values.

Reply:
Addressed by using `minLoss`, `maxGain` in `matrix.layout` and supplying precomputed legend domain, range to `matrix.legend`